pkg/scheduler/preemption/preemption.go

```diff
@@ -334,12 +336,15 @@ func workloadFits(wlReq cache.FlavorResourceQuantities, cq *cache.ClusterQueue,
 			}
 			for rName, rReq := range flvReq {
 				limit := flvQuotas.Resources[rName].Nominal
-				if flvQuotas.Resources[rName].BorrowingLimit != nil && allowBorrowing {
+
+				if cq.Cohort != nil && allowBorrowing && flvQuotas.Resources[rName].BorrowingLimit != nil {
```

Comment:
Based on the comment:
```
	// If null, it means that there is no borrowing limit.
	// If not null, it must be non-negative.
	// borrowingLimit must be null if spec.cohort is empty.
	// +optional
	BorrowingLimit *resource.Quantity `json:"borrowingLimit,omitempty"`
```

If borrowingLimit is nil then cohort must be nil, no bug here. However, we didn't have any validation about this, that's say we can still have cq with `cohort == nil && borrowingLimit != nil`.  

My suggestion here: https://github.com/kubernetes-sigs/kueue/pull/1385#discussion_r1418479788, tl;dr: can we ignore this rule? cc @alculquicondor

Reply:
cc @B1F030